File: lib/qtool.rb
module Qtool

  module DataCreator
    def self.create(amount, max_value)
      data = Array.new
      amount.times do
        data << Random.rand(max_value)
      end
      data
    end
  end

  module DataPreview
    def preview(data)
      preview = []
      data.each_with_index do |element, index|
        preview << [index, element]
      end
      format_preview(preview)
    end

    private

    def format_preview(value)
      value.each do |elements|
        puts "Index: #{elements[0]} - Value: #{elements[1]}"
      end
    end
  end
end

File: lib/qtool.js
exports.dataCreator = {
  create: function(amount, max_value) {
    var data = [];
    for(var index = 0; index < amount; index++) {
      data.push(Math.floor(Math.random() * max_value));
    }
    return data;
  }
};

exports.dataPreview = {
  preview: function(data){
    var preview = [];
    for(var index in data){
      preview.push([index, data[index]]);
    }
    formatPreview(preview);
  }
};

function formatPreview(preview) {
  for(var index in preview) {
    console.log("Index: " + preview[index][0] + " Value: " + preview[index][1] );
  }
};
